### PR TITLE
[clang-tidy] Add AllowImplicitlyDeletedCopyOrMove option to cppcoreguidelines-special-member-functions

### LINKED ISSUE #62392: clang-tidy complains on implicitly deleted special member functions
A common way of disabling copy/move operations is to inherit from a base class that deletes them.

For example:
```
#include <boost/core/noncopyable.hpp>
#include <iostream>

struct Test : private boost::noncopyable {
    ~Test() { std::cout << "dtor\n"; }
};
```

Even though both copy and move operations are implicitly deleted in `Test`, clang-tidy still issues a warning on the destructor:
```
[<source>:4:8: warning: class 'Test' defines a non-default destructor but does not define a copy constructor or a copy assignment operator [cppcoreguidelines-special-member-functions,hicpp-special-member-functions]]
struct Test : private boost::noncopyable {
       ^
```
That's the whole idea of `boost::noncopyable` and similar mixins.

See https://godbolt.org/z/Gohb4Gnjf
or https://godbolt.org/z/17cvYMn1b with a different configuration that complains about move operators as well.

### PR DESCRIPTION
**Pull request by PiotrZSL**
Improved cppcoreguidelines-special-member-functions check with a new option AllowImplicitlyDeletedCopyOrMove, which removes the requirement for explicit copy or move special member functions when they are already implicitly deleted.

Closes #62392

### CODE REVIEW

**Review by 5chmidti — APPROVED** [review]
I think this is correct, but please wait for another approval.

**Comment by PiotrZSL** [comment]
Bump.

**Review by HerrCai0907 — APPROVED** [review]
LGTM

**Pull request merged**
